### PR TITLE
Message stats dashboard

### LINKED ISSUE #3: Views for looking at engagement and attendance metrics

### PR DESCRIPTION
**Pull request by elimelt**
Author: Elijah

## What changes were made?

Added dashboard for viewing discord engagement metrics

### CODE REVIEW

File: src/pages/admin/APIClient.tsx
Comment (elimelt) [inline src/pages/admin/APIClient.tsx:1]: feel free to ignore, this has already been reviewed

File: src/pages/admin/DiscordMessageEngagementDashboardPage.tsx [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        duration: 3000,
+        isClosable: true,
+      });
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const parseMemberIds = (input: string): number[] => {
+    if (!input.trim()) return [];
```
Comment (Advayp) [inline src/pages/admin/DiscordMessageEngagementDashboardPage.tsx]: Maybe move to a separate file?

File: src/pages/admin/DiscordMessageEngagementDashboardPage.tsx [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+      .filter((id) => id !== '')
+      .map((id) => {
+        const num = Number(id);
+        return !isNaN(num) ? num : null;
+      })
+      .filter((id): id is number => id !== null);
+  };
+
+  const parseChannelIds = (input: string): string[] => {
+    if (!input.trim()) return [];
```
Comment (Advayp) [inline src/pages/admin/DiscordMessageEngagementDashboardPage.tsx]: Maybe move to a separate file?

File: src/pages/admin/DiscordMessageEngagementDashboardPage.tsx [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+            <Card variant="filled" bg="whiteAlpha.50">
+              <CardBody>
+                <Heading size="md" mb={4}>
+                  Member Activity
+                </Heading>
+                <Stack spacing={4}>
+                  {aggregatedStats.memberTotals
+                    .sort((a, b) => b.total - a.total)
+                    .map(({ member, total }) => (
+                      <Box key={member.id}>
```
Comment (Advayp) [inline src/pages/admin/DiscordMessageEngagementDashboardPage.tsx]: Maybe extract this `Box` into its own component?

File: src/pages/admin/DiscordMessageEngagementDashboardPage.tsx [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+            <Card variant="filled" bg="whiteAlpha.50">
+              <CardBody>
+                <Heading size="md" mb={4}>
+                  Channel Activity
+                </Heading>
+                <Stack spacing={4}>
+                  {Object.entries(aggregatedStats.channelTotals)
+                    .sort(([, a], [, b]) => b - a)
+                    .map(([channel, count]) => (
+                      <Box key={channel}>
```
Comment (Advayp) [inline src/pages/admin/DiscordMessageEngagementDashboardPage.tsx]: Consider extracting this `Box` into its own component

**Review by Advayp — CHANGES_REQUESTED** [review]

**Review by Advayp — APPROVED** [review]
LGTM

**Pull request merged**